pkg/controller/cluster/cluster_controller_test.go

```diff
@@ -85,42 +85,50 @@ func getKey(cluster *clusteroperator.Cluster, t *testing.T) string {
 	}
 }
 
-// newCluster creates a new cluster with compute node groups that have
+// newCluster creates a new cluster with compute machine sets that have
 // the specified names.
 func newCluster(computeNames ...string) *clusteroperator.Cluster {
-	computes := make([]clusteroperator.ClusterComputeNodeGroup, len(computeNames))
+	computes := make([]clusteroperator.ClusterMachineSet, len(computeNames))
 	for i, computeName := range computeNames {
-		computes[i] = clusteroperator.ClusterComputeNodeGroup{
+		computes[i] = clusteroperator.ClusterMachineSet{
 			Name: computeName,
-			ClusterNodeGroup: clusteroperator.ClusterNodeGroup{
-				Size: 1,
+			Spec: clusteroperator.MachineSetSpec{
+				Size:     1,
+				NodeType: clusteroperator.NodeTypeCompute,
 			},
 		}
 	}
 	return newClusterWithSizes(1, computes...)
 }
 
 // newClusterWithSizes creates a new cluster with the specified size for the
-// master node group and the specified cluster compute node groups.
-func newClusterWithSizes(masterSize int, computes ...clusteroperator.ClusterComputeNodeGroup) *clusteroperator.Cluster {
+// master machine set and the specified cluster compute machine sets.
+func newClusterWithSizes(masterSize int, computes ...clusteroperator.ClusterMachineSet) *clusteroperator.Cluster {
 	cluster := &clusteroperator.Cluster{
 		ObjectMeta: metav1.ObjectMeta{
 			UID:       testClusterUUID,
 			Name:      testClusterName,
 			Namespace: testNamespace,
 		},
 		Spec: clusteroperator.ClusterSpec{
-			MasterNodeGroup: clusteroperator.ClusterNodeGroup{
-				Size: masterSize,
-			},
-			ComputeNodeGroups: computes,
+			MachineSets: append(computes, clusteroperator.ClusterMachineSet{
+				Name: "master",
+				Spec: clusteroperator.MachineSetSpec{
```

Comment:
Should the master machine set have `infra` true to match up with what is in the cluster status?